brooklyn_museum rewritten

Nikitag0329IITkgp opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #[issue number] by @[issue author]

## Description
<!-- Concisely describe what the pull request does. -->

## Technical details
<!-- Add any other information or technical details about the implementation; or delete the section entirely. -->

## Tests
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete the section entirely. -->

## Screenshots
<!-- Add screenshots to show the problem and the solution; or delete the section entirely. -->

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

Commit:
New commit

Nikitag0329IITkgp commented:
Referring to the https://github.com/creativecommons/cccatalog/issues/348, I have rewritten brooklyn_museum.py using DelayedRequester and ImageStore and completely removed eltmods

Nikitag0329IITkgp committed:
Update brooklyn_museum.py

Nikitag0329IITkgp committed:
Update brooklyn_museum.py

kss682 commented:
@Nikitag0329IITkgp The brooklyn Museum API needed to be tested   before rewriting .

akshgpt7 commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,203 @@
+		logging.warning('License not detected!')
+		return None
+
+	title       = sanitizeString(objectData.get('title', ''))
```
The `sanitizeString()` method is a part of the deprecated etlMods and is not required in this script.

Nikitag0329IITkgp replied:
Okay, will look into that. Thanks.

Nikitag0329IITkgp replied:
Please have a look at my updated file @akshgpt7. Thank You

akshgpt7 commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,203 @@
+
+logger = logging.getLogger(__name__)
+
+DELAY       = 3.0 #time delay (in seconds)
```
Please use `pycodestyle` to check for PEP-8 compliance and fix all PEP-8 violations.

Nikitag0329IITkgp replied:
sure. Thanks.

akshgpt7 reviewed (changes requested):
It's better to write smaller functions as it'll make writing tests much easier, when you get to that. 
I'd suggest you to extract logic from the bigger functions and make separate (and smaller) functions for them. 
Please look at `flickr.py` for reference.

kss682 commented:
@Nikitag0329IITkgp Actualy I was planning to work on this , If you are not working could you mention it in the issue so that I can continue working on it.

Nikitag0329IITkgp commented:
> @Nikitag0329IITkgp Actualy I was planning to work on this , If you are not working could you mention it in the issue so that I can continue working on it.

No problem. I will do that.

Nikitag0329IITkgp committed:
Update brooklyn_museum.py

I've removed the usage of sanitizestring, and code style has been formatted according to the pycodestyle.

kss682 commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -36,156 +36,159 @@
     'skip': 0
```
in brooklyn museum offset should be used instead of skip

Nikitag0329IITkgp replied:
okay, I am working on this.

kss682 commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -16,18 +16,18 @@
+image_store = image.ImageStore(provider=PROVIDER, output_file=FILE)
 
 DEFAULT_QUERY_PARAM = {
     'cc': '1',
```
cc is not a parameter for brooklyn

kss682 commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -16,18 +16,18 @@
-ENDPOINT    = 'https://www.brooklynmuseum.org/api/v2/'
+DELAY = 3.0  # time delay (in seconds)
+RETRIES = 3
+FILE = 'brooklynmuseum_{}.tsv'.format(int(time.time()))
```
output file path is not necessary  since the default path is handled by ImageStore

kss682 commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -36,156 +36,159 @@
+
+    while condition:
+        query_param = _build_query_param(offset)
+        endpoint = '{0}object?has_images=1&rights_type_permissive=1&limit={1}&offset={2}'.format(ENDPOINT, LIMIT, offset)
```
move the parameters to the DEFAULT QUERY PARAM

mathemancer replied:
This is correct.  See other scripts (e.g., `flickr.py`) for examples

mathemancer replied:
Actually, many of these are defined in both places (e.g., `DEFAULT_QUERY_PARAM` and in this string.  This will produce unpredictable results.

kss682 reviewed (commented):
Make it more modularize so that testing can be made easier . Like seperate functions to get metadata , license url ,etc. 
Also try running the script locally and verify it . 
Good work

Nikitag0329IITkgp committed:
Update brooklyn_museum.py

Changes: Made function handle_response more modularized by writing functions to get a license, and metadata. Updated Default query params, removed unnecessary FILE path.

Nikitag0329IITkgp commented:
I have made the necessary changes and updated the file but I'm not able to know as to why some checks are failing since it's running locally. Can someone help me out?

kss682 commented:
@Nikitag0329IITkgp The checks is failing because the docker compose is not able to build the testing environment and is failing with
```
Service 'webserver' failed to build: The command '/bin/sh -c pip install --user -r requirements.txt' returned a non-zero code: 2
```
Not exactly sure why this occuring , I suggest you post it in slack so we can know if anyone else has faced this problem or why this is coming.

mathemancer commented:
@Nikitag0329IITkgp Please try pushing an empty commit
```shell
git commit --allow-empty -m "Retrigger Testing Workflow"
git push
```
It seems like the automated testing failed because of some network hiccup, and not due to a problem in your code.

Nikitag0329IITkgp commented:
> @Nikitag0329IITkgp Please try pushing an empty commit
> 
> ```shell
> git commit --allow-empty -m "Retrigger Testing Workflow"
> git push
> ```
> 
> It seems like the automated testing failed because of some network hiccup, and not due to a problem in your code.
 Okay, thank you.

Nikitag0329IITkgp commented:
Okay @mathemancer will do that.

mathemancer committed:
Edit docstring to run automated tests

Nikitag0329IITkgp commented:
@mathemancer can you review this once?

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,216 @@
+def _get_license_url(rightsInfo):
+    if "creative commons" not in rightsInfo.get("name").lower():
+        return None
+    licenseURL = re.search('https://creativecommons.org/licenses/[^\s]+',
```
`re` is undefined here.  You should import the appropriate library.

mathemancer reviewed (changes requested):
This is a good start.  For the most part, the style is readable; thanks for that!
Please double-check the following:
- [ ] Make sure that everything being called is defined.  See my comment regarding `re` below for an example of what I mean.
- [ ] Please remove all unused imports, constants, and variables.
- [ ] Please observe PEP8.  Many things will be caught by `pycodestyle`.  Other problems (which won't) are to make sure that you use `snake_case` rather than `camelCase` or variable names.

Make sure that the file will load, and that all functions can run before re-requesting review.

Thank you!

Nikitag0329IITkgp committed:
Update Brooklyn museum with necessary changes

imported re, removed unused imports, checked with pycodestyle

mathemancer commented:
Is this ready for re-review?  If so, please request a review from me using the Github UI.

Nikitag0329IITkgp commented:
@mathemancer please re-review it. Thanks.

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,214 @@
+LIMIT = 10
+ENDPOINT = 'https://www.brooklynmuseum.org/api/v2/'
+
+logging.basicConfig(format='%(asctime)s: [%(levelname)s - Brooklyn Museum API] =======> %(message)s', level=logging.INFO)
```
Please use the `logging.basicConfig` style from the other new Provider API Scripts, e.g., `flickr.py`.

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,214 @@
+    image_count = 0
+    headers = {'api_key': API_KEY}
+
+    while condition:
```
Please give `condition` a meaningful name.

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,214 @@
+
+
+def _get_meta_data(object_data):
+    meta_data['credit_line'] = object_data.get('credit_line')
+    meta_data['medium'] = object_data.get('medium')
+    meta_data['description'] = object_data.get('description')
+    meta_data['date'] = object_data.get('object_date')
+    meta_data['credit_line'] = object_data.get('period')
+    meta_data['classification'] = object_data.get('classification')
+    meta_data['accession_number'] = object_data.get('accession_number')
+    return meta_data
```
`meta_data` is undefined.

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,214 @@
+        return response_json, total_images
+
+
+def _handle_response(object_data, _object_id):
```
Please break this function into smaller pieces (e.g., refactor it into multiple smaller functions).  It'll make testing easier, and it makes the logic easier to follow and verify.

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,214 @@
+    foreign_url = ''
+    title = ''
+    creator = ''
+    license = ''
```
This variable is assigned but never used.

mathemancer commented on src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py:
```diff
@@ -0,0 +1,214 @@
+
+
+def _handle_response(object_data, _object_id):
+    meta_data = {}
+    img_url = ''
+    width = ''
+    height = ''
+    foreign_id = ''
+    foreign_url = ''
+    title = ''
+    creator = ''
+    license = ''
+    version = ''
```
Please only set these sorts of defaults if there's a good reason (e.g., avoiding multiple/many return statements from the function).

mathemancer reviewed (changes requested):
Please see my comments on specific lines.  In general, 
- [ ] Please make sure there's no long lines (as per PEP8).  
- [ ] Please make sure the script will run before resubmitting.  There are a number of problems with the code that should prevent it from running, such as in the `_get_meta_data` function.

Nikitag0329IITkgp committed:
Update Brooklyn museum

allen505 commented:
Since @kss682 has made a new pull request #355 for the same issue and since @Nikitag0329IITkgp has discontinued work on this as mentioned [here](https://github.com/creativecommons/cccatalog/issues/348#issuecomment-608260152) can this pull request be closed?

mathemancer commented:
@allen505 Thanks for raising the flag.  We'll let @kss682 do the implementation on this.
